File: cs/libldt3/model/objekte/UntersuchungsergebnisZytologieKrebsvorsorge.cs
/*
 * Copyright 2016  Christoph Brill <egore911@gmail.com>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using System.Collections.Generic;
using libldt3.attributes;
using libldt3.model.enums;
using libldt3.model.regel.kontext;

namespace libldt3
{
	namespace model
	{
		namespace objekte
		{

			/**
			 * In diesem Objekt werden die Ergebnisse aus dem Bereich Zytologie
			 * Krebsvorsorge transportiert. Die Inhalte richten sich nach dem Muster 39b.
			 */
			[Objekt(Value = "0062", Kontextregeln = new[] { typeof(K076)})]
			public class UntersuchungsergebnisZytologieKrebsvorsorge
			{

				[Objekt(Kontextregeln = new[] { typeof(K099) })]
				public class TestIdent
				{
					public string Value;
					[Feld(Value = "8411", Feldart = Feldart.bedingt_muss)]
					[Regelsatz(MaxLaenge = 60)]
					public string testbezeichnung;
					[Feld(Value = "8422", Feldart = Feldart.bedingt_muss)]
					[Regelsatz(MaxLaenge = 2)]
					public IList<GrenzwertindikatorErweitert> grenzwertindikator;
					[Feld(Value = "8237", Name = "Ergebnistext", Feldart = Feldart.bedingt_muss)]
					[Regelsatz(Laenge = 12)]
					public Fliesstext ergebnistextVerweis;
				}

				[Feld(Value = "7304", Feldart = Feldart.muss)]
				[Regelsatz(MaxLaenge = 60)]
				public string ergebnisId;
				[Feld(Value = "7364", Feldart = Feldart.muss)]
				[Regelsatz(MaxLaenge = 60)]
				public IList<string> probengefaessIdent;
				[Feld(Value = "8410", Feldart = Feldart.muss)]
				[Regelsatz(MaxLaenge = 60)]
				public TestIdent testIdent;
				[Feld(Value = "8418", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public TestStatus? teststatus;
				[Feld(Value = "7368", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 1)]
				public bool? zellmaterialNichtVerwertbar;
				[Feld(Value = "7405", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public EndozervikaleZellen? endozervikaleZellen;
				[Feld(Value = "7406", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(MaxLaenge = 10)]
				public string proliferationsgrad;
				[Feld(Value = "7407", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public bool? doederleinflora;
				[Feld(Value = "7408", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public bool? mischflora;
				[Feld(Value = "7409", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public bool? kokkenflora;
				[Feld(Value = "7410", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public bool? trichomonaden;
				[Feld(Value = "7411", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public bool? candida;
				[Feld(Value = "7412", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public bool? gardnerella;
				[Feld(Value = "7414", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(MaxLaenge = 5)]
				public string gruppe;
				[Feld(Value = "7413", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(MaxLaenge = 4)]
				public string codierungGruppe;
				[Feld(Value = "7415", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public bool? zytologischeKontrolle;
				[Feld(Value = "7416", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public NachkontrollGrund? nachkontrollGrund;
				[Feld(Value = "7417", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 1)]
				public IList<bool?> histologischeKlaerung;
				[Feld(Value = "8237", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 10 /* XXX 12 according to spec */)]
				public Fliesstext ergebnistext;
				[Feld(Value = "8134", Name = "Krebsfrueherkennung_Frauen", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 26)]
				public KrebsfrueherkennungFrauen krebsfrueherkennungFrauen;
				[Feld(Value = "8126", Name = "Fehlermeldung_Aufmerksamkeit", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 28)]
				public FehlermeldungAufmerksamkeit fehlermeldungAufmerksamkeit;
				[Feld(Value = "8220", Name = "Timestamp_Eingangserfassung_Material", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 36)]
				public Timestamp timestampEingangserfassungMaterial;
				[Feld(Value = "8222", Name = "Timestamp_Beginn_Analytik", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 25)]
				public Timestamp timestampBeginnAnalytik;
				[Feld(Value = "8223", Name = "Timestamp_Ergebniserstellung", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 28)]
				public Timestamp timestampErgebniserstellung;
				[Feld(Value = "8224", Name = "Timestamp_QM_Erfassung", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 22)]
				public Timestamp timestampQmErfassung;
				[Feld(Value = "8225", Name = "Timestamp_Messung", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 17)]
				public Timestamp timestampMessung;
				[Feld(Value = "8167", Name = "Zusaetzliche_Informationen", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 26)]
				public IList<Fliesstext> zusaetzlicheInformationen;
				[Feld(Value = "8110", Feldart = Feldart.kann)]
				[Regelsatz(Laenge = 6)]
				public IList<Anhang> anhang;
				[Feld(Value = "8141", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 13)]
				public Namenskennung namenskennung;
				[Feld(Value = "8158", Feldart = Feldart.bedingt_muss)]
				[Regelsatz(Laenge = 23)]
				public Untersuchungsabrechnung untersuchungsabrechnung;

			}
		}
	}
}

File: java/src/main/java/libldt3/model/objekte/UntersuchungsergebnisZytologieKrebsvorsorge.java
/*
 * Copyright 2016-2022  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.objekte;

import java.util.List;

import libldt3.annotations.Feld;
import libldt3.annotations.Feldart;
import libldt3.annotations.Objekt;
import libldt3.annotations.Regelsatz;
import libldt3.model.enums.EndozervikaleZellen;
import libldt3.model.enums.GrenzwertindikatorErweitert;
import libldt3.model.enums.NachkontrollGrund;
import libldt3.model.enums.TestStatus;
import libldt3.model.regel.kontext.K076;
import libldt3.model.regel.kontext.K099;

/**
 * In diesem Objekt werden die Ergebnisse aus dem Bereich Zytologie
 * Krebsvorsorge transportiert. Die Inhalte richten sich nach dem Muster 39b.
 */
@Objekt(value = "0062", kontextregeln = K076.class)
public class UntersuchungsergebnisZytologieKrebsvorsorge {

	@Objekt(kontextregeln = K099.class)
	public static class TestIdent {
		@SuppressWarnings("unused")
		public String value;
		@Feld(value = "8411", feldart = Feldart.bedingt_muss)
	@Regelsatz(maxLaenge = 60)
		public String testbezeichnung;
		@Feld(value = "8422", feldart = Feldart.bedingt_muss)
	@Regelsatz(maxLaenge = 2)
		public List<GrenzwertindikatorErweitert> grenzwertindikator;
		@Feld(value = "8237", name = "Ergebnistext", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 12)
		public Fliesstext ergebnistextVerweis;
	}

	@Feld(value = "7304", feldart = Feldart.muss)
	@Regelsatz(maxLaenge = 60)
	public String ergebnisId;
	@Feld(value = "7364", feldart = Feldart.muss)
	@Regelsatz(maxLaenge = 60)
	public List<String> probengefaessIdent;
	@Feld(value = "8410", feldart = Feldart.muss)
	@Regelsatz(maxLaenge = 60)
	public TestIdent testIdent;
	@Feld(value = "8418", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public TestStatus teststatus;
	@Feld(value = "7368", feldart = Feldart.kann)
	@Regelsatz(laenge = 1)
	public Boolean zellmaterialNichtVerwertbar;
	@Feld(value = "7405", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public EndozervikaleZellen endozervikaleZellen;
	@Feld(value = "7406", feldart = Feldart.bedingt_muss)
	@Regelsatz(maxLaenge = 10)
	public String proliferationsgrad;
	@Feld(value = "7407", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public Boolean doederleinflora;
	@Feld(value = "7408", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public Boolean mischflora;
	@Feld(value = "7409", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public Boolean kokkenflora;
	@Feld(value = "7410", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public Boolean trichomonaden;
	@Feld(value = "7411", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public Boolean candida;
	@Feld(value = "7412", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public Boolean gardnerella;
	@Feld(value = "7414", feldart = Feldart.bedingt_muss)
	@Regelsatz(maxLaenge = 5)
	public String gruppe;
	@Feld(value = "7413", feldart = Feldart.bedingt_muss)
	@Regelsatz(maxLaenge = 4)
	public String codierungGruppe;
	@Feld(value = "7415", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public Boolean zytologischeKontrolle;
	@Feld(value = "7416", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public NachkontrollGrund nachkontrollGrund;
	@Feld(value = "7417", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 1)
	public List<Boolean> histologischeKlaerung;
	@Feld(value = "8237", feldart = Feldart.kann)
	@Regelsatz(laenge = 10 /* XXX 12 according to spec */)
	public Fliesstext ergebnistext;
	@Feld(value = "8134", name = "Krebsfrueherkennung_Frauen", feldart = Feldart.kann)
	@Regelsatz(laenge = 26)
	public KrebsfrueherkennungFrauen krebsfrueherkennungFrauen;
	@Feld(value = "8126", name = "Fehlermeldung_Aufmerksamkeit", feldart = Feldart.kann)
	@Regelsatz(laenge = 28)
	public FehlermeldungAufmerksamkeit fehlermeldungAufmerksamkeit;
	@Feld(value = "8220", name = "Timestamp_Eingangserfassung_Material", feldart = Feldart.kann)
	@Regelsatz(laenge = 36)
	public Timestamp timestampEingangserfassungMaterial;
	@Feld(value = "8222", name = "Timestamp_Beginn_Analytik", feldart = Feldart.kann)
	@Regelsatz(laenge = 25)
	public Timestamp timestampBeginnAnalytik;
	@Feld(value = "8223", name = "Timestamp_Ergebniserstellung", feldart = Feldart.kann)
	@Regelsatz(laenge = 28)
	public Timestamp timestampErgebniserstellung;
	@Feld(value = "8224", name = "Timestamp_QM_Erfassung", feldart = Feldart.kann)
	@Regelsatz(laenge = 22)
	public Timestamp timestampQmErfassung;
	@Feld(value = "8225", name = "Timestamp_Messung", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 17)
	public Timestamp timestampMessung;
	@Feld(value = "8167", name = "Zusaetzliche_Informationen", feldart = Feldart.kann)
	@Regelsatz(laenge = 26)
	public List<Fliesstext> zusaetzlicheInformationen;
	@Feld(value = "8110", feldart = Feldart.kann)
	@Regelsatz(laenge = 6)
	public List<Anhang> anhang;
	@Feld(value = "8141", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 13)
	public Namenskennung namenskennung;
	@Feld(value = "8158", feldart = Feldart.bedingt_muss)
	@Regelsatz(laenge = 23)
	public Untersuchungsabrechnung untersuchungsabrechnung;

}
